File: Week2/basic/cpp/W_AllYouNeedIsLove.cpp
#include <bits/stdc++.h>
#define int long long
using namespace std;

int gcd(int a, int b) {
    return b == 0 ? a : gcd(b, a % b);
}

void solve() {
    string s1, s2; cin >> s1 >> s2;
    int n1 = stoll(s1, 0, 2), n2 = stoll(s2, 0, 2);

    int g = gcd(n1, n2);

    if (g == 1) cout << "Love is not all you need!\n";
    else cout << "All you need is love!\n";
}

signed main() {
    ios::sync_with_stdio(0); cin.tie(0);
    int T; cin >> T;
    for (int t = 1; t <= T; t++) {
        cout << "Pair #" << t << ": ";
        solve();
    }
}

File: Week2/basic/Java/W_AllYouNeedIsLove.java
import java.util.*;

public class W_AllYouNeedIsLove {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int i = 1; i <= T; i++) {
            String s1 = sc.next();
            String s2 = sc.next();

            // 十進位轉二進位
            long a = Long.parseLong(s1, 2);
            long b = Long.parseLong(s2, 2);

            // 找最大公因數
            long c = gcd(a, b);
            
            if (c == 1) System.out.printf("Pair #%d: Love is not all you need!\n", i);
            else System.out.printf("Pair #%d: All you need is love!\n", i);
        }
        sc.close();
    }

    public static long gcd(long a, long b) {
        return b == 0 ? a : gcd(b, a % b);
    }
}
